### ISSUE TITLE
Avoid clearing color where possible

### ISSUE THREAD

**Issue by mourner** [mapbox/mapbox-gl-js#8651]
From [article on GPU performance](https://developer.nvidia.com/gpugems/GPUGems/gpugems_ch28.html):

> Avoid extraneous color-buffer clears. If every pixel is guaranteed to be overwritten in the frame buffer by your application, then avoid clearing color, because it costs precious bandwidth.

In almost all cases, Mapbox styles have a non-transparent background layer — in case we detect one, we can avoid calling `gl.clear({color})` on every frame. cc @ansis what do you think?

**Comment by ansis** [comment]
I think our rendering would still work correctly if we dropped the colour clear in cases we have a solid background. But I'm not sure this is the best idea: https://stackoverflow.com/questions/37335281/is-glcleargl-color-buffer-bit-preferred-before-a-whole-frame-buffer-overwritte

@kkaefer might now more about this

We used to use gl clearing for drawing solid backgrounds at some point but I can't remember why that changed. Maybe to get depth clipping? Not sure.
